### PR TITLE
Issue 6219: Additional Segment Store Health-checks

### LINKED ISSUE #6219: Segment store health
**Is your feature request related to a problem? Please describe.**
We are currently unable to tell the details related to health of the segment store. It would be useful to get details on why a segment store could be potentially failing.

**Describe the solution you'd like**
HealthChecks for segment store with details about the health.

**Additional context**
Add health details regarding zookeeper client connection, pravega connection listener, admin connection listener, cache manager and container list.

### PR DESCRIPTION
**Pull request by dellThejas**
Add healthcheck features for segment store.

Signed-off-by: dellThejas <Thejas.Vidyasagar@dell.com>

**Change log description**  
Added additional healtcheck for the following features in segmentstore:

- cache manager
- container registry

**Purpose of the change**  
Fixes #6219 .

**What the code does**  
Added healthContributors for features in segmentstore:

- cache manager
- container registry

**How to verify it**  
Run pravega and run `curl -v http:localhost:controllerRestPort` and `curl -v http:localhost:segmentStoreRestPort`

### CODE REVIEW

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/health/SegmentContainerHealthContributor.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    public SegmentContainerHealthContributor(@NonNull SegmentContainer segmentContainer) {
+        super("SegmentContainer");
+        this.segmentContainer = segmentContainer;
+    }
+
+    @Override
+    public Status doHealthCheck(Health.HealthBuilder builder) {
+        Status status = Status.DOWN;
+
+        boolean newState = segmentContainer.state() == Service.State.NEW;
```
Comment (RaulGracia) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/health/SegmentContainerHealthContributor.java:42]: Why you create a new `boolean` variable every time? Wouldn't it be equivalent to do something like:
```
if (segmentContainer.state() == Service.State.NEW) {
    status = Status.NEW;
} else if (segmentContainer.state() == Service.State.STARTING) {
    status = Status.STARTING;
} else if (segmentContainer.state() == Service.State.RUNNING) {
    status = Status.UP;
}
```

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/health/SegmentContainerHealthContributor.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import com.google.common.util.concurrent.Service;
+import io.pravega.segmentstore.server.SegmentContainer;
+import io.pravega.shared.health.Health;
+import io.pravega.shared.health.Status;
+import io.pravega.shared.health.impl.AbstractHealthContributor;
+import lombok.NonNull;
+
+
+/**
+ *  A contributor to manage health of segment container.
```
Comment (RaulGracia) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/health/SegmentContainerHealthContributor.java:28]: Nit: Normally, Pravega concepts like "Segment Container" start by capitals.

File: segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/health/SegmentContainerRegistryHealthContributor.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+import io.pravega.segmentstore.server.SegmentContainer;
+import io.pravega.segmentstore.server.SegmentContainerRegistry;
+import io.pravega.shared.health.Health;
+import io.pravega.shared.health.Status;
+import io.pravega.shared.health.impl.AbstractHealthContributor;
+import lombok.NonNull;
+
+/**
+ * A contributor to manage the health of segment container registry.
```
Comment (RaulGracia) [inline segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/health/SegmentContainerRegistryHealthContributor.java:27]: Nit: same here "Segment Container Registry"

File: segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/ServiceStarterTest.java [hunk 4]
```diff
@@ -80,6 +85,18 @@ public void testCuratorClientCreation() throws Exception {
...
+    /**
+     * Check health of SegmentContainerRegistry
+     */
+    @Test
+    public void testSegmentContainerRegistryHealth() {
+        @Cleanup
+        SegmentContainerRegistryHealthContributor segmentContainerRegistryHealthContributor = new SegmentContainerRegistryHealthContributor(serviceStarter.getServiceBuilder().getSegmentContainerRegistry());
+        Health.HealthBuilder builder = Health.builder().name(segmentContainerRegistryHealthContributor.getName());
+        Status status = segmentContainerRegistryHealthContributor.doHealthCheck(builder);
+        Assert.assertEquals("HealthContributor should report an 'UP' Status.", Status.UP, status);
```
Comment (RaulGracia) [inline segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/ServiceStarterTest.java:97]: Can you test also the transition to "DOWN" state?

**Review by RaulGracia — CHANGES_REQUESTED** [review]
Looks good, just few minor comments.

File: segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/ServiceStarterTest.java (hunk 4 above)
Comment (dellThejas) [reply segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/ServiceStarterTest.java:97]: It cant be tested here. It is being tested here instead: `SegmentContainerRegistryHealthContributorTest`

**Review by RaulGracia — APPROVED** [review]

**Review by andreipaduroiu — APPROVED** [review]

**Pull request merged**